scipy/stats/tests/test_morestats.py

```diff
@@ -1211,7 +1211,7 @@ def test_exact_basic(self):
         for n in range(1, 26):
             cnt = _get_wilcoxon_distr(n)
             assert_equal(n*(n+1)/2 + 1, len(cnt))
-            assert_equal(sum(cnt), 2**n)
+            assert_equal(sum(cnt), 1)
```

Comment:
If we think more tests are needed, you could test `_get_wilcoxon_distr` against this implementation, which does the same thing but is based on [this reference](https://journals.sagepub.com/doi/pdf/10.1177/1536867X1301300208).

```
from scipy.fft import ifft

def _get_wilcoxon_distr2(n):
    ai = np.arange(1, N+1)[:, None]
    T = N*(N+1)/2
    Q = 2*T
    j = np.arange(Q)
    theta = 2*np.pi/Q*j
    Phi_Sp = np.prod(np.cos(theta*ai), axis=0)  # this is slow
    Phi_S = np.exp(1j*theta*T) * Phi_Sp
    p = np.real(ifft(Phi_S))
    res = np.zeros(int(T)+1)
    res[:-1:] = p[::2]
    res[0] /= 2
    res[-1] = res[0]
    return res
```

I thought it might be faster. It is not. Supposedly it can be used to calculate the exact value of the test statistic with corrections for ties, though.